birdhouse/components/weaver/post-docker-compose-up

```diff
@@ -268,37 +283,53 @@ for prov in ${WEAVER_WPS_PROVIDERS}; do
-    echo "${PREFIX}Unregistering any remote WPS provider matching [${prov}]."
-    curl_cmd --insecure --silent --location \
-         -m ${REQUEST_TIMEOUT} \
-         -w "${PREFIX}Delete [${prov}] response: %{http_code}" -o /dev/null \
-         -b "${cookie}" \
-         -X DELETE \
-         "${WEAVER_URL}/providers/${prov}"
-
-    # register the new provider and validate
-    printf "\n%s" "${PREFIX}Registering remote WPS provider [${prov}] on [${prov_url}]... "
-    resp=$( \
+    retry=0
+    retry_msg=""
+    total=${WEAVER_WPS_PROVIDERS_RETRY_COUNT}
+    while true; do
+        if [ ${retry} -ne 0 ]; then
+          retry_msg=" (retry: ${retry}/${total})"
+        fi
+        # unregister in case of multiple up/down to regenerate from scratch, don't care if NotFound returned
+        echo "${PREFIX}Unregistering any remote WPS provider matching [${prov}]${retry_msg}."
         curl_cmd --insecure --silent --location \
              -m ${REQUEST_TIMEOUT} \
-             -w "\n%{http_code}" \
+             -w "${PREFIX}Delete [${prov}] response: %{http_code}${retry_msg}" -o /dev/null \
              -b "${cookie}" \
-             -H "Content-Type: application/json" \
-             -X POST \
-             -d "{\"id\": \"${prov}\", \"url\": \"${prov_url}\"}" \
-             "${WEAVER_URL}/providers" \
-    )
-    ret=$?
-    code=$(echo "${resp}" | tail -n -1)
-    body=$(echo "${resp}" | head -n -1)
-    if [ ${ret} -ne 0 ] || [ "${code}" -ne 201 ]; then
-        printf "\n%s\n" "${ERROR}Failed registration of remote WPS provider [${prov}] on [${prov_url}]."
-        printf "Error:\n%s\n" "${body}"
-        reset_state
-        exit 23
-    fi
-    echo "OK!"
+             -X DELETE \
+             "${WEAVER_URL}/providers/${prov}"
+
+        # register the new provider and validate
+        printf "\n%s" "${PREFIX}Registering remote WPS provider [${prov}] on [${prov_url}]${retry_msg}... "
+        resp=$( \
+            curl_cmd --insecure --silent --location \
+                 -m ${REQUEST_TIMEOUT} \
+                 -w "\n%{http_code}" \
+                 -b "${cookie}" \
+                 -H "Content-Type: application/json" \
+                 -X POST \
+                 -d "{\"id\": \"${prov}\", \"url\": \"${prov_url}\"}" \
+                 "${WEAVER_URL}/providers" \
+        )
+        ret=$?
+        code=$(echo "${resp}" | tail -n -1)
+        body=$(echo "${resp}" | head -n -1)
+        if [ ${ret} -ne 0 ] || [ "${code}" -ne 201 ]; then
+            printf "\n%s\n" "${WARN}Failed registration of remote WPS provider [${prov}] on [${prov_url}]${retry_msg}."
+            printf "Error:\n%s\n" "${body}"
+            if [ ${retry} -gt ${total} ]; then
+              echo "${ERROR}Maximum retry attempts ${total} reached for WPS provider [${prov}]. Aborting."
+              reset_state
+              exit 23
+            fi
+            echo "${WARN}Will retry after ${WEAVER_WPS_PROVIDERS_RETRY_AFTER}s..."
```

Comment:
For messages displayed to the user, I think it's better to actually use the word "seconds". As in:

```
Will retry after 5 seconds
```

instead of 

```
Will retry after 5s
```

Reply:
I'll leave it as is because it aligns with all other outputs reporting seconds and keeps the message more succinct.
It's only for debugging anyway.